exercises/concept/card-tricks/card_tricks_test.go

```diff
@@ -133,121 +132,141 @@ func TestSetItem(t *testing.T) {
 				t.Errorf("SetItem(slice:%v, index:%v, value:%v) = %v, want %v",
 					tt.args.slice, tt.args.index, tt.args.value, got, tt.want)
 			}
+			if len(tt.args.slice) == len(got) {
+				for i := range got {
+					got[i] = -1
+				}
+				if !slicesEqual(got, tt.args.slice) {
+					t.Errorf("SetItem(slice:%v, index:%v) does not return the modified input slice)", tt.args.slice,
+						tt.args.value)
+				}
+			}
 		})
 	}
 }
 
-func TestPrefilledSlice(t *testing.T) {
+func TestRemoveItem(t *testing.T) {
 	type args struct {
-		value  int
-		length int
+		slice []int
+		index int
 	}
 	tests := []struct {
 		name string
 		args args
 		want []int
 	}{
 		{
-			name: "Create a prefilled slice with value 3",
+			name: "Remove an item",
+			args: args{
+				slice: []int{3, 4, 5, 6},
+				index: 1,
+			},
+			want: []int{3, 5, 6},
+		},
+		{
+			name: "Remove the first item",
 			args: args{
-				value:  3,
-				length: 7,
+				slice: []int{3, 4, 5, 6},
+				index: 0,
 			},
-			want: []int{3, 3, 3, 3, 3, 3, 3},
+			want: []int{4, 5, 6},
 		},
 		{
-			name: "Create a prefilled slice with value 10",
+			name: "Remove the last item",
 			args: args{
-				value:  10,
-				length: 2,
+				slice: []int{3, 4, 5, 6},
+				index: 3,
 			},
-			want: []int{10, 10},
+			want: []int{3, 4, 5},
 		},
 		{
-			name: "Length zero",
+			name: "Remove out of bounds index",
 			args: args{
-				value:  3,
-				length: 0,
+				slice: []int{3, 4, 5, 6},
+				index: 7,
 			},
-			want: []int{},
+			want: []int{3, 4, 5, 6},
 		},
 		{
-			name: "Negative length",
+			name: "Remove negative index",
 			args: args{
-				value:  3,
-				length: -3,
+				slice: []int{3, 4, 5, 6},
+				index: -7,
 			},
-			want: []int{},
+			want: []int{3, 4, 5, 6},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := PrefilledSlice(tt.args.value, tt.args.length); !slicesEqual(got, tt.want) {
-				t.Errorf("PrefilledSlice(value:%v, length:%v) = %v, want %v", tt.args.value, tt.args.length, got, tt.want)
+			if got := RemoveItem(copySlice(tt.args.slice), tt.args.index); !slicesEqual(got, tt.want) {
+				t.Errorf("RemoveItem(slice:%v, index:%v) = %v, want %v", tt.args.slice, tt.args.index, got, tt.want)
 			}
 		})
 	}
 }
 
-func TestRemoveItem(t *testing.T) {
+func TestPrependItems(t *testing.T) {
 	type args struct {
 		slice []int
-		index int
+		value []int
 	}
 	tests := []struct {
 		name string
 		args args
 		want []int
 	}{
 		{
-			name: "Remove an item",
+			name: "prepend nil",
 			args: args{
-				slice: []int{3, 4, 5, 6},
-				index: 1,
+				slice: []int{5, 2, 10, 6, 8, 7, 0, 9},
+				value: nil,
 			},
-			want: []int{3, 5, 6},
+			want: []int{5, 2, 10, 6, 8, 7, 0, 9},
 		},
 		{
-			name: "Remove the first item",
+			name: "prepend zero items",
 			args: args{
-				slice: []int{3, 4, 5, 6},
-				index: 0,
+				slice: []int{5, 2, 10, 6, 8, 7, 0, 9},
+				value: []int{},
 			},
-			want: []int{4, 5, 6},
+			want: []int{5, 2, 10, 6, 8, 7, 0, 9},
```

Comment:
I would recommend to move these edge cases to the end of the test list so the student sees a usual example in the test case first.

Reply:
Done.